doc: replace `-?` with `-h` and `-help`

l0rinc opened the pull request:
The question mark is interpreted as a wildcard for any single character in Zsh (see https://www.techrepublic.com/article/globbing-wildcard-characters-with-zsh), so `bench_bitcoin -?` will not show the help message on systems using Zsh, such as macOS.

Since `-h` provides equivalent help functionality (as defined in https://github.com/bitcoin/bitcoin/blob/master/src/common/args.cpp#L684-L693), the `benchmarking.md` documentation has been updated to ensure compatibility with macOS. 

----

### -?
> % cmake -B build -DBUILD_BENCH=ON && cmake --build build && build/src/bench/bench_bitcoin -?
zsh: no matches found: -?

### -h
> % cmake -B build -DBUILD_BENCH=ON && cmake --build build && build/src/bench/bench_bitcoin -h
Usage:  bench_bitcoin [options]
Options:
...

----

Based on the comments the args help default was also changed to `-help`, mentioning `-h` and `-?` (instead of `-?` being the default)

l0rinc committed:
doc: replace `-?` with `-h` for bench_bitcoin help

The question mark (`?`) is interpreted as a wildcard for any single character in Zsh (see https://www.techrepublic.com/article/globbing-wildcard-characters-with-zsh), so `bench_bitcoin -?` will not work on systems using Zsh, such as macOS.

Since `-h` provides equivalent help functionality (as defined in https://github.com/bitcoin/bitcoin/blob/master/src/common/args.cpp#L684-L693), the `benchmarking.md` documentation has been updated to ensure compatibility with macOS.
\

laanwj reviewed (approved):
ACK f0130ab1a1e65583637b6a362b879ea3253e7bb7
Easier to remember too, `-?` is a kind of mingling between Windows/DOS `/?` and UNIX `-h` conventions.

rkrux reviewed (approved):
tACK https://github.com/bitcoin/bitcoin/commit/f0130ab1a1e65583637b6a362b879ea3253e7bb7

I compiled `bench_bitcoin`, and ran the help command on Mac OS. `-h` works, `-?` doesn't.

Wondering why is `-h` a hidden argument. Do you think it's worth making it non-hidden with its own message just like `-?`?
https://github.com/bitcoin/bitcoin/blob/master/src/common/args.cpp#L692

laanwj commented:
> Wondering why is -h a hidden argument. Do you think it's worth making it non-hidden with its own message just like -??

There's no reason to list both seperately. That would clutter the list. `-?` is the weird duck out in that it lists the short-name option as primary instead of the long one (`-help`). Might add the alternatives to the help message say,

```
  -help
       Print this help message and exit (also -h or -?)
```

BrandonOdiwuor reviewed (approved):
ACK f0130ab1a1e65583637b6a362b879ea3253e7bb7, great catch

rkrux commented:
> > Wondering why is -h a hidden argument. Do you think it's worth making it non-hidden with its own message just like -??
> 
> There's no reason to list both seperately. That would clutter the list. `-?` is the weird duck out in that it lists the short-name option as primary instead of the long one (`-help`). Might add the alternatives to the help message say,
> 
> ```
>   -help
>        Print this help message and exit (also -h or -?)
> ```

@laanwj Yes, this^ seems better.

l0rinc committed:
Change default help arg to `-help` and mention `-h` and `-?` as alternatives

% build/src/bench/bench_bitcoin -h
[...]
  -help
       Print this help message and exit (also -h or -?)

l0rinc commented:
I've pushed the default change in a separate commit, let me know if this is what you had in mind:
```bash
% build/src/bench/bench_bitcoin -h
[...]
  -help
       Print this help message and exit (also -h or -?)
```

rkrux reviewed (approved):
tACK https://github.com/bitcoin/bitcoin/commit/33a28e252a7349c0aa284005aee97873b965fcfe

Thanks @l0rinc for addressing the change quickly. Rebuilt and checked for bitcoin-cli as well, it's updated there too.

```
➜  bitcoin git:(l0rinc/bench-help) ✗ bitcoincli -h
Bitcoin Core RPC client version v28.99.0-33a28e252a73

Usage:  bitcoin-cli [options] <command> [params]  Send command to Bitcoin Core
or:     bitcoin-cli [options] -named <command> [name=value]...  Send command to Bitcoin Core (with named arguments)
or:     bitcoin-cli [options] help                List commands
or:     bitcoin-cli [options] help <command>      Get help for a command

Options:

  -color=<when>
       Color setting for CLI output (default: auto). Valid values: always, auto
       (add color codes when standard output is connected to a terminal
       and OS is not WIN32), never.

  -conf=<file>
       Specify configuration file. Relative paths will be prefixed by datadir
       location. (default: bitcoin.conf)

  -datadir=<dir>
       Specify data directory

  -help
       Print this help message and exit (also -h or -?)
```

laanwj reviewed (approved):
Code review ACK 33a28e252a7349c0aa284005aee97873b965fcfe

hebasto commented:
Concept ACK.

edilmedeiros commented:
Concept ACK

edilmedeiros commented:
tACK 33a28e252a7349c0aa284005aee97873b965fcfe

I would prefer everything in a single commit, but it's OK.

fanquake commented on doc/benchmarking.md:
```diff
@@ -40,7 +40,7 @@ The output will look similar to:
 ---------------------
 
-    build/src/bench/bench_bitcoin -?
+    build/src/bench/bench_bitcoin -h
```
Maybe this will be the last time we rewrite this.
Introduced as `-?`: b21244e0be5727f8c4e8c5de0a9aa2c597ae8ed2
Changed to `--help`: 05fdb97df46d0a0675b93e9791bd5d498e5e5117
Changed back to `-?`: d8513fe41102dcbfc05235f3b95e33eb1878f880
Now we are going back to `-h`: f0130ab1a1e65583637b6a362b879ea3253e7bb7.

l0rinc replied:
I'm fine with `--help` as well, but `-?` just isn't working on a mac

achow101 commented:
ACK 33a28e252a7349c0aa284005aee97873b965fcfe

hebasto reviewed (commented):
Post-merge ACK 33a28e252a7349c0aa284005aee97873b965fcfe.
